### PR TITLE
add group-add flag for run

### PR DESCRIPTION
**Pull request by junnplus**
Signed-off-by: Ye Sijun <junnplus@gmail.com>

### CODE REVIEW

File: cmd/nerdctl/run_user.go [hunk 1]
```diff
@@ -42,3 +50,70 @@ func withResetAdditionalGIDs() oci.SpecOpts {
 		return nil
 	}
 }
+
+// Copy from https://github.com/containerd/containerd/blob/v1.6.6/oci/spec_opts.go#L1298-L1303
+func tryReadonlyMounts(mounts []mount.Mount) []mount.Mount {
+	if len(mounts) == 1 && mounts[0].Type == "overlay" {
```
Comment (AkihiroSuda) [inline cmd/nerdctl/run_user.go:56]: What happens for non-overlay filesystems?

cc @fuweid

File: cmd/nerdctl/run_user.go [hunk 2]
```diff
@@ -42,3 +50,70 @@ func withResetAdditionalGIDs() oci.SpecOpts {
...
+	}
+	addlGids := []uint32{}
+	for _, grp := range groups {
+		addlGids = append(addlGids, uint32(grp.Gid))
+	}
+	return addlGids, nil
+}
+
+// withAdditionalGIDs adds additional gids.
+func withAdditionalGIDs(groups ...string) oci.SpecOpts {
```
[the remark this answers is not in the document]
Comment (junnplus) [reply cmd/nerdctl/run_user.go:82]: SGTM, that would be better.

[the remark this answers is not in the document]
Comment (junnplus) [reply cmd/nerdctl/run_user.go:63]: https://github.com/containerd/containerd/pull/7070

File: cmd/nerdctl/run_user_linux_test.go [hunk 3]
```diff
@@ -52,3 +52,69 @@ func TestRunUmask(t *testing.T) {
...
+			t.Parallel()
+			cmd := []string{"run", "--rm"}
+			if testCase.user != "" {
+				cmd = append(cmd, "--user", testCase.user)
+			}
+			for _, group := range testCase.groups {
+				cmd = append(cmd, "--group-add", group)
+			}
+			cmd = append(cmd, testutil.AlpineImage, "id", "-nG")
+			base.Cmd(cmd...).AssertOutContains(testCase.expected)
```
Comment (AkihiroSuda) [inline cmd/nerdctl/run_user_linux_test.go:117]: Can we use `AssertOutExactly` ?

Comment (junnplus) [reply cmd/nerdctl/run_user_linux_test.go]: Done.

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**